src/dmd/typesem.d

```diff
@@ -3624,6 +3624,23 @@ Expression dotExp(Type mt, Scope* sc, Expression e, Identifier ident, int flag)
             }
             e = e.castTo(sc, e.type.nextOf().pointerTo());
         }
+        else if (ident == Id._tupleof)
+        {
+            if (e.op == EXP.type)
+            {
+                e.error("Can only use .tupleof on static array values (perhaps you meant `typeof(%s.init.tupleof)`?)", mt.toChars);
```

Comment:
```suggestion
                e.error("Cannot use `.tupleof` on type `%s`, mt.toChars);
                e.errorSupplemental("Did you mean `typeof(%s.init.tupleof)`?", mt.toChars());
```

IMO this is more in consistent with other messages.

Reply:
Cannot use -> .tupleof cannot be used on